File: app/src/main/java/com/jqorz/planewar/entity/BombSupply.java
package com.jqorz.planewar.entity;

import com.jqorz.planewar.base.BaseSupply;
import com.jqorz.planewar.tools.BitmapLoader;
import com.jqorz.planewar.tools.MapCreator;
import com.jqorz.planewar.utils.ConstantUtil;

/**
 * 该类为炸弹补给类
 */
public class BombSupply extends BaseSupply {


    BombSupply() {
        super(BitmapLoader.bmp_bombSupply);
    }

    @Override
    public void reset() {
        setX(MapCreator.getNewSupplyPos(this));
        setY(-getHeight());
        setShown(true);
    }

    @Override
    public void move() {
        y += ConstantUtil.Bomb_Velocity;
    }

}

File: app/src/main/java/com/jqorz/planewar/entity/BombSupply.kt
package com.jqorz.planewar.entity

import com.jqorz.planewar.base.BaseSupply
import com.jqorz.planewar.constant.ConstantValue
import com.jqorz.planewar.tools.BitmapLoader
import com.jqorz.planewar.tools.MapCreator

/**
 * 该类为炸弹补给类
 */
class BombSupply internal constructor() : BaseSupply(BitmapLoader.bmp_bombSupply) {
    override fun reset() {
        setX(MapCreator.getNewSupplyPos(this))
        setY(-height)
        setShown(true)
    }

    override fun move() {
        y += ConstantValue.Bomb_Velocity
    }
}